src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/handlers/presubmit/EditBailApplicationAfterSubmitHandler.java

```diff
@@ -0,0 +1,88 @@
+import uk.gov.hmcts.reform.bailcaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.bailcaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.bailcaseapi.domain.entities.ccd.field.IdValue;
+import uk.gov.hmcts.reform.bailcaseapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.bailcaseapi.domain.service.DocumentReceiver;
+import uk.gov.hmcts.reform.bailcaseapi.domain.service.DocumentsAppender;
+
+@Component
+public class EditBailApplicationAfterSubmitHandler implements PreSubmitCallbackHandler<BailCase> {
+
+    private final DocumentReceiver documentReceiver;
+    private final DocumentsAppender documentsAppender;
+
+    public EditBailApplicationAfterSubmitHandler(
+        DocumentReceiver documentReceiver,
+        DocumentsAppender documentsAppender
+    ) {
+        this.documentReceiver = documentReceiver;
+        this.documentsAppender = documentsAppender;
+    }
+
+    @Override
+    public DispatchPriority getDispatchPriority() {
+        return DispatchPriority.EARLIEST;
+    }
+
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<BailCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+               && callback.getEvent() == Event.EDIT_BAIL_APPLICATION_AFTER_SUBMIT;
+    }
+
+    public PreSubmitCallbackResponse<BailCase> handle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<BailCase> callback
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final BailCase bailCase =
+            callback
+                .getCaseDetails()
+                .getCaseData();
+
+        final Optional<List<IdValue<DocumentWithMetadata>>> maybeApplicantDocuments =
+            bailCase.read(APPLICANT_DOCUMENTS_WITH_METADATA);
+
+        if (maybeApplicantDocuments.isPresent()) {
+            List<IdValue<DocumentWithMetadata>> applicantDocumentsList = maybeApplicantDocuments.get();
+
+            List<IdValue<DocumentWithMetadata>> updatedApplicantDocumentsList =
+                applicantDocumentsList
+                .stream()
+                .filter(documentWithMetaData ->
+                            !documentWithMetaData.getValue().getTag().equals(DocumentTag.B1_DOCUMENT))
+                .filter(documentWithMetaData ->
+                            !documentWithMetaData.getValue().getTag().equals(DocumentTag.BAIL_EVIDENCE))
+                .filter(documentWithMetaData ->
+                            !documentWithMetaData.getValue().getTag().equals(DocumentTag.BAIL_SUBMISSION))
+                .collect(Collectors.toList());
```

Comment:
I believe this could be synthesized like this:
```
List<IdValue<DocumentWithMetadata>> updatedApplicantDocumentsList =
                applicantDocumentsList
                .stream()
                .filter(documentWithMetaData ->
                            !documentWithMetaData.getValue().getTag().equals(DocumentTag.B1_DOCUMENT)
                            || !documentWithMetaData.getValue().getTag().equals(DocumentTag.BAIL_EVIDENCE)
                            || !documentWithMetaData.getValue().getTag().equals(DocumentTag.BAIL_SUBMISSION))
                .collect(Collectors.toList());
```
But it would need to be checked that is works, I only wrote it up

Reply:
I had a quick check - it looks like it isn't able to pick up the variable documentWithMetaData after the or operator.

Reply:
Good shout, this has been added in now in the latest commit.